src/app/courses-page/course-list/course-list.component.ts

```diff
@@ -0,0 +1,41 @@
+import { Component, OnInit } from '@angular/core';
+
+import { CourseData } from '../../data-models/course-data';
+import { FilterPipe } from '../../pipes/filter.pipe';
+import { CourseService } from '../../services/course.service';
+
+@Component({
+  selector: 'app-course-list',
+  templateUrl: './course-list.component.html',
+  styleUrls: ['./course-list.component.scss']
+  //providers: [FilterPipe]
+})
+export class CourseListComponent implements OnInit {
+  public courses: CourseData[] = [];
+  public noDataMessage: string = 'No data. Feel free to add new course';
+
+  constructor(
+    private _filterPipe: FilterPipe,
+    private _courseService: CourseService ) { }
+
+  ngOnInit(): void {
+    this.courses = this._courseService.getCourseList();
+  }
+
+  trackByCourseId(index: number, course: any) {
```

Comment:
don'y gorget to add types
 **public** trackByCourseId(index: number, course: any): **number** {